RUFAS/biophysical/animal/animal.py

```diff
@@ -1,2 +1,2409 @@
+from RUFAS.biophysical.animal.digestive_system.digestive_system import DigestiveSystem
+from RUFAS.biophysical.animal.growth.growth import Growth
+from RUFAS.biophysical.animal.nutrients.nutrients import Nutrients
+from RUFAS.biophysical.animal.nutrients.nasem_requirements_calculator import NASEMRequirementsCalculator
+from RUFAS.biophysical.animal.nutrients.nrc_requirements_calculator import NRCRequirementsCalculator
+from RUFAS.biophysical.animal.data_types.animal_statistics import AnimalStatistics
+from RUFAS.biophysical.animal.data_types.animal_typed_dicts import (
+    NewBornCalfValuesTypedDict,
+    CalfValuesTypedDict,
+    HeiferIValuesTypedDict,
+    HeiferIIValuesTypedDict,
+    CowValuesTypedDict,
+    HeiferIIIValuesTypedDict,
+)
+from RUFAS.biophysical.animal.data_types.animal_types import AnimalType
+from RUFAS.biophysical.animal.data_types.repro_protocol_enums import (
+    HeiferReproductionProtocol,
+    HeiferTAISubProtocol,
+    HeiferSynchEDSubProtocol, CowReproductionProtocol, CowPreSynchSubProtocol, CowTAISubProtocol, CowReSynchSubProtocol,
+)
+from RUFAS.biophysical.animal.milk.lactation_curve import LactationCurve
+from RUFAS.biophysical.animal.milk.milk_production import MilkProduction
+from RUFAS.biophysical.animal.ration.amino_acid import EssentialAminoAcidRequirements
+from RUFAS.biophysical.animal.ration.calf_ration_manager import CalfRationManager
+from RUFAS.biophysical.animal.reproduction.reproduction import Reproduction
+from RUFAS.data_structures.feed_storage_to_animal_connection import NutrientStandard, Feed
+from RUFAS.general_constants import GeneralConstants
+from RUFAS.time import Time
+
+
 class Animal:
-    pass
+    """
+    DO NOT USE THE PROPERTIES THAT START WITH '_'. INSTEAD, USE THE FUNCTIONS THAT ARE DECORATED WITH @property.
+
+    Attributes
+    ----------
+    id: int
+        The unique identifier of the animal, (unitless).
+    breed: Breed
+        The breed of the animal.
+    animal_type: AnimalType
+        The current life stage of the animal.
+    days_born: int
+        The age of the animal, (simulation days).
+    body_weight: float
+        The body weight of the animal, (kg).
+    birth_weight: float
+        The birth weight of the animal, (kg).
+    mature_body_weight: float
+        The mature body weight of the animal, (kg).
+    wean_weight: float
+        The body weight of the animal at weaning, (kg).
+    net_merit: float
+        The net merit value of the animal, ($USD).
+    body_condition_score_5: float
+        The body condition score on a scale of 1 to 5, (unitless).
+    cull_reason: str
+        The reason for the animal to leave the herd.
+    body_weight_history: list[BodyWeightHistory]
+        The body weight history of the animal.
+    pen_history: list[PenHistory]
+        The pen history of the animal.
+    sold_at_day: int | None
+        The simulation day in which the animal was sold, (simulation day).
```

Comment:
Not quite so sure that simulation day is a unit, if not perhaps we can remove it?
```suggestion
        The simulation day in which the animal was sold.
```